plugins/org.jkiss.dbeaver.ext.altibase.ui/src/org/jkiss/dbeaver/ext/altibase/ui/config/AltibaseTypesetConfigurator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.altibase.ui.config;
+
+import org.jkiss.dbeaver.ext.altibase.model.AltibaseTypeset;
+import org.jkiss.dbeaver.ext.altibase.ui.views.CreateTypesetPage;
+import org.jkiss.dbeaver.model.edit.DBEObjectConfigurator;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.struct.rdb.DBSProcedureType;
+import org.jkiss.dbeaver.ui.UITask;
+import org.jkiss.dbeaver.utils.GeneralUtils;
+
+import java.util.Map;
+
+/**
+ * AltibaseProcedureConfigurator
+ */
+public class AltibaseTypesetConfigurator implements DBEObjectConfigurator<AltibaseTypeset> {
+
+    @Override
+    public AltibaseTypeset configureObject(DBRProgressMonitor monitor, Object container, 
+            AltibaseTypeset procedure, Map<String, Object> options) {
+        return new UITask<AltibaseTypeset>() {
+            @Override
+            protected AltibaseTypeset runTask() {
+                CreateTypesetPage editPage = new CreateTypesetPage(procedure);
+                if (!editPage.edit()) {
+                    return null;
+                }
+                DBSProcedureType procedureType = editPage.getProcedureType();
+                String procedureName = editPage.getProcedureName();
+
+                procedure.setName(procedureName);
+                procedure.setProcedureType(procedureType);
+
+                procedure.setObjectDefinitionText(
+                    "CREATE OR REPLACE TYPESET " + procedureName +
+                    (procedureType == DBSProcedureType.FUNCTION ? "() RETURN NUMBER" : "") + GeneralUtils.getDefaultLineSeparator() +
+                        "AS" + GeneralUtils.getDefaultLineSeparator() +
+                        "\t" + GeneralUtils.getDefaultLineSeparator() + 

```

Comment:
In summary, it will look like an empty row. And it will not work because an empty line is a statement separator by default in DBeaver.

Reply:
It returns a DDL skeleton to create a new typeset DDL with user provided name. If I misunderstood your point, please let me know.

![image](https://github.com/dbeaver/dbeaver/assets/32251385/848fb05a-c389-45bd-8005-fad18172d2f9)
![image](https://github.com/dbeaver/dbeaver/assets/32251385/52803865-d20e-4277-86d2-1fa61eaa8bdf)
![image](https://github.com/dbeaver/dbeaver/assets/32251385/7352ada9-b7a1-4a58-bf56-9bfadd47d91d)

Reply:
Here is my question.